pkg/cmd/cli/cmd/env.go

```diff
@@ -260,7 +265,10 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 				return nil
 			}
 			for _, c := range containers {
-				c.Env = updateEnv(c.Env, env, remove)
+				c.Env, envUpdated = updateEnv(c.Env, env, remove)
+				if envUpdated {
```

Comment:
but in this case it doesn't matter which wins. Once the `updated[i]` is set to `true` it can't be changed in any iteration.